q2_cutadapt/plugin_setup.py

```diff
@@ -32,6 +45,150 @@
                   'pp. 10-12.\ndoi:http://dx.doi.org/10.14806/ej.17.1.200',
 )
 
+plugin.methods.register_function(
+    function=q2_cutadapt._trim.trim_single,
+    inputs={
+        'demultiplexed_sequences': SampleData[SequencesWithQuality],
+    },
+    parameters={
+        'cores': Int % Range(1, None),
+        'adapter': List[Str],
+        'front': List[Str],
+        'anywhere': List[Str],
+        'error_rate': Float % Range(0, 1, inclusive_start=True,
+                                    inclusive_end=True),
+        'indels': Bool,
+        'times': Int % Range(1, None),
+        'overlap': Int % Range(1, None),
+        'match_read_wildcards': Bool,
+        'match_adapter_wildcards': Bool,
+    },
+    outputs=[
+        ('trimmed_sequences', SampleData[SequencesWithQuality]),
+    ],
+    input_descriptions={
+        'demultiplexed_sequences': 'The single-end sequences to be trimmed.',
+    },
+    parameter_descriptions={
+        'cores': 'Number of CPU cores to use.',
```

Comment:
Is there a value to pass to indicate all cores? If so, could you note that here?

Reply:
Not to my knowledge (at least, I didn't see anything in the docs for this).

`# TODO: add missing parallel support libs to the conda recipe;`